File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        collection = {}

        for i in s:
            if i in collection:
                collection[i] += 1
            else:
                collection[i] = 1
        
        for j in t:
            if j not in collection:
                return False
            else:
                collection[j] -= 1
        
        for k in collection:
            if collection[k] != 0:
                return False

        return True

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    let collection = {}

    for (var i in s) {
        if (s[i] in collection) {
            collection[s[i]] += 1
        } else {
            collection[s[i]] = 1
        }
    }
    
    for (var j in t) {
        if (!t[j] in collection) {
            return false
        } else {
                collection[t[j]] -= 1
        }
    }

    for (var k in collection) {
        if (collection[k] != 0) {
            return false;
        }
    }

    return true;
    
};